osbs/build/build_request.py

```diff
@@ -98,22 +100,34 @@ def set_params(self, **kwargs):
 
         # Here we cater to the koji "scratch" build type, this will disable
         # all plugins that might cause importing of data to koji
-        try:
-            self.scratch = kwargs.pop("scratch")
-        except KeyError:
-            pass
+        self.scratch = kwargs.pop('scratch', False)
+        # When true, it indicates build was automatically started by
+        # OpenShift via a trigger, for instance ImageChangeTrigger
+        self.is_auto = kwargs.pop('is_auto', False)
+        # An isolated build is meant to patch a certain release and not
+        # update transient tags in container registry
+        self.isolated = kwargs.pop('isolated', False)
+
+        self.validate_build_variation()
 
         self.base_image = kwargs.get('base_image')
         self.platform_node_selector = kwargs.get('platform_node_selector', {})
         self.platform_descriptors = kwargs.get('platform_descriptors', {})
         self.scratch_build_node_selector = kwargs.get('scratch_build_node_selector', {})
         self.explicit_build_node_selector = kwargs.get('explicit_build_node_selector', {})
         self.auto_build_node_selector = kwargs.get('auto_build_node_selector', {})
-        self.is_auto = kwargs.get('is_auto', False)
 
         logger.debug("setting params '%s' for %s", kwargs, self.spec)
         self.spec.set_params(**kwargs)
 
+    def validate_build_variation(self):
+        variations = (self.scratch, self.is_auto, self.isolated)
+        enabled_variations = list(filter(lambda x: x, variations))
```

Comment:
This `filter` looks useless to me, what does it do?

Reply:
What about
```
variations = (self.scratch, self.is_auto, self.isolated)
if variations.count(True) > 1:
    raise ValueError('Build variations are mutually exclusive. '
                               'Must set either scratch, is_auto, isolated, or none. ')
```?

Reply:
Sure, that would produce a similar result.